### PR TITLE
Docs: Add acknowledgements

### PR DESCRIPTION
**Pull request by bgopi23**
Added acknowledgements for me and @dillontkh

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -10,13 +10,22 @@ title: Developer Guide
...
+| [Jackson](https://github.com/FasterXML/jackson) | Json processing library for Java          |
+| [JUnit5](https://github.com/junit-team/junit5)  | Automated testing library for Java        |
 | [ZXing](https://github.com/zxing/zxing)         | Barcode image processing library for Java |
 
+**AI Tools Acknowledgements**
+
+| Tool    | Name            | Extent of use                             |
+|---------|-----------------|-------------------------------------------|
+| GPT-4   | Baskar Gopinath | JavaDocs (FitAddCommandParser)            |
+| Codeium | Dillon          | <li> JavaDocs <li> Full-line autocomplete |
```
Comment (dillontkh) [inline docs/DeveloperGuide.md:27]: Before:
image

After:
image


```suggestion
| Codeium | Dillon          | {::nomarkdown}<ul><li>JavaDocs</li><li>Full-line autocomplete</li></ul>{:/} |
```

**Review by dillontkh — CHANGES_REQUESTED** [review]
Thanks! Just 1 markdown fix required.

**Review by dillontkh — APPROVED** [review]
LGTM!

**Review by jjchee77 — APPROVED** [review]
LGTM!

**Review by RyanNgWH — APPROVED** [review]
LGTM!

**Pull request merged**